### PR TITLE
ENH: use native filesystem (if available) for read_parquet with pyarrow engine

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Some timings in https://issues.apache.org/jira/browse/ARROW-12428 suggest that it can be beneficial for reading parquet from S3 to use the pyarrow filesystem implementation, instead of pandas always using fsspec if installed (especially when selecting columns).

### CODE REVIEW

File: pandas/io/parquet.py [hunk 1]
```diff
@@ -172,9 +172,25 @@ def write(
 
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
 
+        filesystem = kwargs.pop("filesystem", None)
```
Comment (jreback) [inline pandas/io/parquet.py:175]: is there a reason we are not actually exposing `filesystem` as a named keyword?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by TomAugspurger** [comment]
Might want to document this behavior. If so, here's my docs from the geopandas PR:

```
        When no storage options are provided and a filesystem is implemented by
        both ``pyarrow.fs`` and ``fsspec`` (e.g. "s3://") then the ``pyarrow.fs``
        filesystem is preferred. Provide the instantiated fsspec filesystem using
        the ``filesystem`` keyword if you wish to use its implementation.
```

**Comment by jreback** [comment]
this is quite old, happen to reopen if actively worked on.

File: pandas/io/parquet.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/io/parquet.py:176]: > is there a reason we are not actually exposing `filesystem` as a named keyword?

That could also be an option. I think the problem is that right now, this is a keyword that is just passed through to the underlying engine (as we do for other engine-specific keywords as well), and `filesystem` is not actually a keyword supported by fastparquet.

Comment (jorisvandenbossche) [reply pandas/io/parquet.py:176]: Actually, to be more correct, fastparquet supports it but names it `fs`. So we could indeed add a keyword `filesystem`, and map it to `fs` for fastparquet?

Comment (jreback) [reply pandas/io/parquet.py:176]: yes this sounds reasonable

**Review by jreback — CHANGES_REQUESTED** [review]
@jorisvandenbossche i fyou can rebase and move the note to 1.5

**Comment by simonjayhawkins** [comment]
closing as stale

**Pull request closed without merging**